services/api/db/serializers.py

```diff
@@ -0,0 +1,109 @@
+from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
+from django.contrib.auth.tokens import PasswordResetTokenGenerator
+from django.core.validators import validate_email
+
+
+class SignUpEndPointSerializer(serializers.ModelSerializer):
+    password2 = serializers.CharField(style={'input_type': 'passowrd'}, write_only=True)
+    class Meta:
+        model = User
+        fields= ['email', 'name', 'password', 'password2']
+        extra_kwargs={
+            'password': {'write_only': True}
+        }
+
+    #validating if password entered are correct.
+    def validate(self, data):
+        email = data.get('email')
+        password, password2 = data.get('password'), data.get('password2')
+        try:
+            validate_email(email)
+        except:
+            raise serializers.ValidationError("Please provide a valid email address.")
+        
+        if password != password2:
+            raise serializers.ValidationError("Password and confirm password doesn't match.")
+        return data
+        
+    def create(self, validated_data):
+        return User.objects.create_user(**validated_data)
+    
+class SignInEndPointSerializer(serializers.ModelSerializer):
+    email = serializers.EmailField(max_length=255)
+    class Meta:
+        model = User
+        fields = ['email', 'password']
+
+class UserProfileSerialiser(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ['id', 'email', 'name']
+
+class UserChangePasswordSerialiser(serializers.Serializer):
+
+    #new password type can be customised here.
+    password = serializers.CharField(max_length=255, style={'input_type' : 'passowrd'}, write_only=True)
+    password2 = serializers.CharField(max_length=255, style={'input_type' : 'passowrd'}, write_only=True)
+    class Meta:
+        fields = ['passowrd', 'password2']
+    
+    def validate(self, attrs):
+        password = attrs.get('password')
+        password2 = attrs.get('password2')
+        user = self.context.get('user')
+        if password != password2:
+            raise serializers.ValidationError("Password and Confirm Password doesn't match")
+        user.set_password(password)
+        user.save()
+        return attrs
+
+    
+class SendPasswordResetEmailSerializer(serializers.Serializer):
+    email = serializers.EmailField(max_length=255)
+    class Meta:
+        fields = ['email']
+
+    def validate(self, attrs):
+        email = attrs.get('email')
+        if User.objects.filter(email=email).exists():
+            user = User.objects.get(email = email)
+            uid = urlsafe_base64_encode(force_bytes(user.id))
+            #print('Encoded UID', uid)
+            token = PasswordResetTokenGenerator().make_token(user)
+            #print('Password Reset Token', token)
+            #Instead of local server, desired domain server could be added here.
+            link = 'http://localhost:3000/api/user/reset/'+uid+'/'+token
```

Comment:
We can use Python F Strings instead of string concatenation.

Reply:
changed into f string